pkg/backfill/backfill.go

```diff
@@ -15,24 +15,18 @@ import (
 )
 
 type Backfill struct {
-	conn       db.DB
-	batchSize  int
-	batchDelay time.Duration
-	callbacks  []CallbackFn
+	conn   db.DB
+	config *Config
```

Comment:
The `config` struct could be embedded: 

```suggestion
   *Config
```

I don't mind either way.